test/testDataEmbeds.js

```diff
@@ -0,0 +1,149 @@
+ */
+/* eslint-env mocha */
+const assert = require('assert');
+const { dom: { assertEquivalentNode } } = require('@adobe/helix-shared');
+const { logging } = require('@adobe/helix-testutils');
+const nock = require('nock');
+const { JSDOM } = require('jsdom');
+const { pipe } = require('../src/defaults/html.pipe.js');
+const coerce = require('../src/utils/coerce-secrets');
+const Downloader = require('../src/utils/Downloader.js');
+
+const params = {
+  path: '/hello.md',
+  __ow_method: 'get',
+  owner: 'trieloff',
+  __ow_headers: {
+    'X-Forwarded-Port': '443',
+    'X-CDN-Request-Id': '2a208a89-e071-44cf-aee9-220880da4c1e',
+    'Fastly-Client': '1',
+    'X-Forwarded-Host': 'runtime.adobe.io',
+    'Upgrade-Insecure-Requests': '1',
+    Host: 'controller-a',
+    Connection: 'close',
+    'Fastly-SSL': '1',
+    'X-Request-Id': 'RUss5tPdgOfw74a68aNc24FeTipGpVfW',
+    'X-Branch': 'master',
+    'Accept-Language': 'en-US, en;q=0.9, de;q=0.8',
+    'X-Forwarded-Proto': 'https',
+    'Fastly-Orig-Accept-Encoding': 'gzip',
+    'X-Varnish': '267021320',
+    DNT: '1',
+    'X-Forwarded-For':
+      '192.147.117.11, 157.52.92.27, 23.235.46.33, 10.64.221.107',
+    'X-Host': 'www.primordialsoup.life',
+    Accept:
+      'text/html, application/xhtml+xml, application/xml;q=0.9, image/webp, image/apng, */*;q=0.8',
+    'X-Real-IP': '10.64.221.107',
+    'X-Forwarded-Server': 'cache-lcy19249-LCY, cache-iad2127-IAD',
+    'Fastly-Client-IP': '192.147.117.11',
+    'Perf-Br-Req-In': '1529585370.116',
+    'X-Timer': 'S1529585370.068237,VS0,VS0',
+    'Fastly-FF':
+      'dc/x3e9z8KMmlHLQr8BEvVMmTcpl3y2YY5y6gjSJa3g=!LCY!cache-lcy19249-LCY, dc/x3e9z8KMmlHLQr8BEvVMmTcpl3y2YY5y6gjSJa3g=!LCY!cache-lcy19227-LCY, dc/x3e9z8KMmlHLQr8BEvVMmTcpl3y2YY5y6gjSJa3g=!IAD!cache-iad2127-IAD, dc/x3e9z8KMmlHLQr8BEvVMmTcpl3y2YY5y6gjSJa3g=!IAD!cache-iad2133-IAD',
+    'Accept-Encoding': 'gzip',
+    'User-Agent':
+      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 Safari/537.36',
+  },
+  repo: 'soupdemo',
+  ref: 'master',
+  selector: 'md',
+};
+
+const secrets = {
+  REPO_RAW_ROOT: 'https://raw.githubusercontent.com/',
+  EMBED_WHITELIST: '*.youtube.com',
+  DATA_EMBED_WHITELIST: 'docs.google.com',
+};
+
+const logger = logging.createTestLogger({
+  // tune this for debugging
+  level: 'debug',
+});
+
+
+const crequest = {
+  extension: 'html',
+  url: '/test/test.html',
+};
+
+describe('Integration Test with Data Embeds', () => {
+  afterEach(() => {
+    nock.restore();
+  });
+
+  beforeEach(() => {
+    nock.restore();
+    nock.activate();
+    nock.cleanAll();
+  });
+
+  it('html.pipe processes data embeds', async () => {
+    /*
+    nock('https://raw.githubusercontent.com')
+      .get('/adobe/test-repo/master/fstab.yaml')
+      .reply(() => 404);
+    */
+
+    console.log('nocking!');
+    nock(/.*/)
+      .get()
+      .reply(404, (uri, requestBody) => {
+        // this never happens
```

Comment:
@tripodsan @stefan-guggisberg Any idea how I can get `nock` to intercept requests made by our `Downloader`? I tried taking the code from`testPipeWithFstab.js`, but the console statement below never gets executed.

Reply:
yes, set: `process.env. HELIX_PIPELINE_FORCE_HTTP1 = true':

https://github.com/adobe/helix-cli/blob/08c46de0399a40d82157e4560b821954fa4a6ecc/test/testUpCmd.js#L63

Reply:
hmm.... this should work...

Reply:
because you have an error in your pipe, and fetch is never executed:

```
<html><body><h1>500</h1><pre>TypeError: handlefn is not a function</pre></body></html>
```

Reply:
It must be something different. The `handlefn` error occurs after the `fetch` step, which is being executed, because I can see the response from runtime.

Reply:
the problem was that you used `unist-util-map` which can't deal with async callbacks.
I added a promisified version:
https://github.com/adobe/helix-pipeline/pull/611/commits/13dca796da8a7f72e794a71af735a7dcf4f82f60

now it's nocking :-)